runtime/v1/linux/runtime.go

```diff
@@ -362,6 +364,22 @@ func (r *Runtime) loadTasks(ctx context.Context, ns string) ([]*Task, error) {
 
 		logDirPath := filepath.Join(r.root, ns, id)
 
+		copyAndClose := func(dst io.Writer, src io.ReadWriteCloser) {
+			copyDone := make(chan struct{})
+			go func() {
+				io.Copy(dst, src)
+				close(copyDone)
+			}()
+			select {
+			case <-shimExit:
+				break
+			case <-copyDone:
+				break
+			}
+			if src != nil {
```

Comment:
I think we can remove the nil check here, because the scope is clear here. And the if src is nil, the inner goroutine will panic.